storage/remote/otlptranslator/prometheus/normalize_label.go

```diff
@@ -28,14 +28,16 @@ import (
 // Labels that start with non-letter rune will be prefixed with "key_".
 //
 // An exception is made for double-underscores which are allowed.
-func NormalizeLabel(label string) string {
+func NormalizeLabel(label string, allowUTF8 bool) string {
 	// Trivial case
 	if len(label) == 0 {
 		return label
 	}
 
-	// Replace all non-alphanumeric runes with underscores
-	label = strings.Map(sanitizeRune, label)
+	if !allowUTF8 {
+		// Replace all non-alphanumeric runes with underscores
+		label = strings.Map(sanitizeRune, label)
+	}
 
 	// If label starts with a number, prepend with "key_"
```

Comment:
Is this required with full utf8 support? Looking at the validation of labels, it doesn't seem like so: https://github.com/prometheus/common/blob/dae848db5327d2a4e2e06cbe883093a71b4226d7/model/metric.go#L164-L176

fwiw the [proposal](https://github.com/prometheus/proposals/pull/28/files) is not very explicit about this

Reply:
Sorry, I meant specifically the adding of `key_` prefix when it starts with a number. That's still being done when full utf8 is enabled.

Reply:
I'm not sure if starting with a number is a UTF8 problem. SDKs also don't allow it, I just tried adding a label `0test` to one of client_golang's example

```
panic: descriptor Desc{fqName: "http_request_duration_seconds", help: "A histogram of the HTTP request durations in seconds.", constLabels: {}, variableLabels: {}} is invalid: "0test" is not a valid label name for metric "http_request_duration_seconds"

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(...)
        /Users/arthursens/Documents/projetos/client_golang/prometheus/registry.go:406
main.main()
        /Users/arthursens/Documents/projetos/client_golang/examples/createdtimestamps/main.go:37 +0x3c0
```

I don't know the reason exactly to be honest. If I have to guess, maybe at some point we face a conversion issue when ingesting into the TSDB...?

Reply:
I would guess it's mostly about PromQL syntax (e.g. `1e3` is a number and not a metric name). 

> . SDKs also don't allow it, I just tried adding a label 0test to one of client_golang's example

Interesting, is that with full UTF8 enabled? Not sure who's right, but that looks inconsistent to what [this method is doing](https://github.com/prometheus/common/blob/dae848db5327d2a4e2e06cbe883093a71b4226d7/model/metric.go#L164-L176)

Reply:
I just got this working locally:

```go
func main() {
	// Notice these lines below
	model.NameValidationScheme = model.UTF8Validation
	model.NameEscapingScheme = model.NoEscaping

	counter := prometheus.NewCounterVec(prometheus.CounterOpts{
		Name: "1the.counter",
		Help: "This is my counter",
	}, []string{"3foo"})

	prometheus.MustRegister(counter)
	counter.WithLabelValues("bar").Inc()

	http.Handle("/metrics", promhttp.Handler())
	http.ListenAndServe(":2112", nil)
}
```

And scraped (running `prom/prometheus:v3.0.0-beta.0`):

![image](https://github.com/user-attachments/assets/8bbfe0fb-abdf-47cd-98ec-928ae340b787)

So it looks like, at least currently, full UTF8 support allows for numbers at the start of metric and label names. 

If we want to use full UTF8 support to leave labels as they come, maybe we should skip the `key_` prefixing  if full UTF8 is enabled? I'm not saying we _have_ to, I'm not very familiar with the Prometheus OTLP receivers/translators. Just bringing it to attention :slightly_smiling_face: .

Reply:
Nice! Thanks for the exploration work :) 

Yeah, if it works then I agree we should do that!